### ISSUE TITLE
Augmentation dot placement

### ISSUE THREAD

**Issue by craigsapp** [rism-digital/verovio#244]
In this graphical rendering:

![screen shot 2016-07-17 at 8 58 24 am](https://cloud.githubusercontent.com/assets/3487289/16901584/c3378888-4bfc-11e6-9abd-673d3df68eba.png)

The augmentation dot on the first note of the bottom layer is in the wrong staff space.  It should be  below the note rather than above due to the occurrence of two voices; otherwise, the augmentation dot looks like it belongs to the upper voice:

![screen shot 2016-07-17 at 8 57 13 am](https://cloud.githubusercontent.com/assets/3487289/16901591/02767dd8-4bfd-11e6-97fd-1bc96b3a36db.png)

I think I also reported this subissue, but since it is in this example as well... also notice the eighth-note rest in the verovio rendering.  It is moved out of the way for no reason.  There is nothing in the other layer which would overlap with it.

Here is the original Humdrum content:

```
**kern
*clefF4
*M3/4
*k[b-e-]
=1-
8r
*^
(8FL    [4.F
8A      .
8cJ     .
4e-)    4F]
*v      *v
=
*-
```

MEI translation:

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-07-17 09:04:35</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-7fa44b3</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000127298478">
                        <staffGrp xml:id="m-000000090619956">
                            <staffDef xml:id="staffdef-000000054198039" clef.shape="F" clef.line="4" key.sig="2f" meter.count="3" meter.unit="4" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure xml:id="measure-000000066244177" n="1">
                            <staff xml:id="staff-000000189063718" n="1">
                                <!-- kern: =1-  8r  *^  (8FL  8A  8cJ  4e-)  *v  = -->
                                <layer xml:id="layer-000000177119451" n="1">
                                    <rest xml:id="rest-000000004783166" dur="8" />
                                    <beam xml:id="beam-000000074787606">
                                        <note xml:id="note-000000033120495" dur="8" oct="3" pname="f" accid.ges="n" />
                                        <note xml:id="note-000000028398162" dur="8" oct="3" pname="a" accid.ges="n" />
                                        <note xml:id="note-000000117052454" dur="8" oct="4" pname="c" accid.ges="n" />
                                    </beam>
                                    <note xml:id="note-000000205587171" dur="4" oct="4" pname="e" accid.ges="f" />
                                </layer>
                                <!-- kern: [4.F  4F]  *v -->
                                <layer xml:id="layer-000000049120603" n="2">
                                    <space xml:id="space-000000077274159" dur="8" />
                                    <note xml:id="note-000000163442417" dots="1" dur="4" oct="3" pname="f" accid.ges="n" />
                                    <note xml:id="note-000000130381406" dur="4" oct="3" pname="f" accid.ges="n" />
                                </layer>
                            </staff>
                            <slur xml:id="slur-000000002405058" staff="1" startid="#note-000000033120495" endid="#note-000000205587171" />
                            <tie xml:id="tie-000000027980604" startid="note-000000163442417" endid="note-000000130381406" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
Here is another example:

<img width="299" alt="screen shot 2017-01-21 at 12 33 52" src="https://cloud.githubusercontent.com/assets/3487289/22177600/f574e260-dfd5-11e6-81a2-c7b70189b646.png">

which should be:

<img width="281" alt="screen shot 2017-01-21 at 12 34 35" src="https://cloud.githubusercontent.com/assets/3487289/22177601/01c334b8-dfd6-11e6-8c9d-7d9468ffb19c.png">

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2017-01-21 12:33:16</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.14-dev-7f64096</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000069338927">
                        <staffGrp xml:id="m-000000047891305" symbol="brace" barthru="true">
                            <staffDef xml:id="staffdef-000000153760689" clef.shape="G" clef.line="2" key.sig="1f" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000028119423" clef.shape="F" clef.line="4" key.sig="1f" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-000000138440538">
                        <measure xml:id="measure-L4" n="19">
                            <staff xml:id="staff-L4F2" n="1">
                                <layer xml:id="layer-L4F2" n="1">
                                    <beam xml:id="beam-000000062610816">
                                        <note xml:id="note-L6F3" dur="16" oct="5" pname="c" accid.ges="n">
                                            <artic xml:id="artic-000000072031809" artic="stacc" />
                                        </note>
                                        <note xml:id="note-L7F3" dur="16" oct="5" pname="d" accid.ges="n">
                                            <artic xml:id="artic-000000080842600" artic="stacc" />
                                        </note>
                                        <note xml:id="note-L8F3" dur="16" oct="5" pname="e" accid.ges="n">
                                            <artic xml:id="artic-000000005853639" artic="stacc" />
                                        </note>
                                        <note xml:id="note-L9F3" dur="16" oct="5" pname="f" accid.ges="n">
                                            <artic xml:id="artic-000000120855952" artic="stacc" />
                                        </note>
                                    </beam>
                                    <note xml:id="note-L10F3" dots="1" dur="4" oct="4" pname="f" accid.ges="n" />
                                    <beam xml:id="beam-000000008201668">
                                        <note xml:id="note-L12F3" dur="16" oct="4" pname="a" accid.ges="n" />
                                        <note xml:id="note-L13F3" dur="16" oct="4" pname="g" accid.ges="n" />
                                    </beam>
                                </layer>
                                <layer xml:id="layer-L6F4" n="2">
                                    <note xml:id="note-L6F4" dur="8" oct="4" pname="a" accid.ges="n" />
                                    <rest xml:id="rest-L8F4" dur="8" />
                                    <note xml:id="note-L10F4" dots="1" dur="4" oct="4" pname="c" accid.ges="n" />
                                    <note xml:id="note-L12F4" dur="8" oct="4" pname="e" accid.ges="n" />
                                </layer>
                            </staff>
                            <staff xml:id="staff-L4F1" n="2">
                                <layer xml:id="layer-L4F1" n="1">
                                    <chord xml:id="chord-L6F1" dur="8">
                                        <note xml:id="note-L6F1S1" oct="2" pname="f" accid.ges="n" />
                                        <note xml:id="note-L6F1S2" oct="3" pname="f" accid.ges="n" />
                                    </chord>
                                    <rest xml:id="rest-L8F1" dur="8" />
                                    <note xml:id="note-L10F1" dur="4" oct="3" pname="a" accid.ges="n" />
                                    <note xml:id="note-L11F1" dur="4" oct="3" pname="b" accid.ges="f" />
                                </layer>
                                <layer xml:id="layer-L6F2" n="2">
                                    <space xml:id="space-L6F2" dur="4" />
                                    <note xml:id="note-L10F2" dur="2" oct="3" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                            <dynam xml:id="dynam-000000122659146" staff="1" tstamp="2.000000">p</dynam>
                            <slur xml:id="slur-000000022750913" staff="2" startid="#note-L10F1" endid="#note-L11F1" />
                            <slur xml:id="slur-000000172901804" staff="1" startid="#note-L10F3" endid="#note-L16F3" />
                        </measure>
                        <measure xml:id="measure-L15" n="20">
                            <staff xml:id="staff-L15F2" n="1">
                                <layer xml:id="layer-L15F2" n="1">
                                    <note xml:id="note-L16F2" dur="4" oct="4" pname="f" accid.ges="n" />
                                    <rest xml:id="rest-L17F2" dur="8" />
                                </layer>
                                <layer xml:id="layer-L15F3" n="2">
                                    <note xml:id="note-L16F3" dur="4" oct="4" pname="f" accid.ges="n" />
                                    <space xml:id="space-L17F3" dur="8" />
                                </layer>
                            </staff>
                            <staff xml:id="staff-L15F1" n="2">
                                <layer xml:id="layer-L15F1" n="1">
                                    <chord xml:id="chord-L16F1" dur="4">
                                        <note xml:id="note-L16F1S1" oct="3" pname="f" accid.ges="n" />
                                        <note xml:id="note-L16F1S2" oct="3" pname="a" accid.ges="n" />
                                    </chord>
                                    <note xml:id="note-L17F1" dur="8" oct="2" pname="f" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
Here is a more extreme example:

<img width="319" alt="screen shot 2017-01-21 at 19 25 58" src="https://cloud.githubusercontent.com/assets/3487289/22179673/91d7a618-e00f-11e6-99ef-526ae7cceaa8.png">

which should look like this:

<img width="270" alt="screen shot 2017-01-21 at 19 27 05" src="https://cloud.githubusercontent.com/assets/3487289/22179676/9d17099c-e00f-11e6-9da7-dc0d6a5a5c2b.png">

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2017-01-21 19:25:54</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.14-dev-ca447f7</p>
            </projectDesc>
        </encodingDesc>
        <extMeta>
            <frames xmlns:humxml="http://www.humdrum.org/ns/humxml">
                <metaFrame n="20" token="!!!RDF**kern: &lt; = below" xml:id="loc20">
                    <frameInfo>
                        <startTime float="3" />
                        <frameType>reference</frameType>
                        <referenceKey>RDF**kern</referenceKey>
                        <referenceValue>&lt; = below</referenceValue>
                    </frameInfo>
                </metaFrame>
                <metaFrame n="21" token="!!!RDF**kern: &gt; = above" xml:id="loc21">
                    <frameInfo>
                        <startTime float="3" />
                        <frameType>reference</frameType>
                        <referenceKey>RDF**kern</referenceKey>
                        <referenceValue>&gt; = above</referenceValue>
                    </frameInfo>
                </metaFrame>
            </frames>
        </extMeta>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000115475668" midi.bpm="88">
                        <staffGrp xml:id="m-000000069777494" symbol="brace" barthru="true">
                            <staffDef xml:id="staffdef-000000075417435" clef.shape="G" clef.line="2" key.sig="1f" meter.count="3" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000007642178" clef.shape="F" clef.line="4" key.sig="1f" meter.count="3" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-000000157700721">
                        <measure xml:id="measure-L7" n="58">
                            <staff xml:id="staff-L7F2" n="1">
                                <layer xml:id="layer-L7F2" n="1">
                                    <space xml:id="space-L9F2" dots="1" dur="4" />
                                    <beam xml:id="beam-000000073197632">
                                        <note xml:id="note-L12F2" dur="8" oct="4" pname="g" accid.ges="n" />
                                        <note xml:id="note-L14F2" dur="8" oct="4" pname="a" accid.ges="n" />
                                        <note xml:id="note-L16F2" dur="8" oct="4" pname="b" accid.ges="f" />
                                    </beam>
                                </layer>
                                <layer xml:id="layer-L9F3" n="2">
                                    <beam xml:id="beam-000000143447858">
                                        <note xml:id="note-L9F3" dots="1" dur="8" oct="5" pname="c" accid.ges="n" />
                                        <note xml:id="note-L10F3" dur="16" accid="n" oct="4" pname="b" />
                                    </beam>
                                    <note xml:id="note-L11F3" dur="8" accid="f" oct="4" pname="b" />
                                    <beam xml:id="beam-000000045586999">
                                        <note xml:id="note-L12F3" dots="1" dur="16" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L13F3" dur="32" oct="4" pname="c" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000078121219">
                                        <note xml:id="note-L14F3" dots="1" dur="16" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L15F3" dur="32" oct="4" pname="c" accid.ges="n" />
                                    </beam>
                                    <beam xml:id="beam-000000025845566">
                                        <note xml:id="note-L16F3" dots="1" dur="16" oct="4" pname="g" accid.ges="n" />
                                        <note xml:id="note-L17F3" dur="32" oct="4" pname="c" accid.ges="n" />
                                    </beam>
                                </layer>
                            </staff>
                            <staff xml:id="staff-L7F1" n="2">
                                <layer xml:id="layer-L7F1" n="1">
                                    <chord xml:id="chord-L9F1" dots="1" dur="4">
                                        <note xml:id="note-L9F1S1" oct="3" pname="e" accid.ges="n" />
                                        <note xml:id="note-L9F1S2" oct="3" pname="g" accid.ges="n" />
                                        <note xml:id="note-L9F1S3" oct="4" pname="c" accid.ges="n" />
                                    </chord>
                                    <beam xml:id="beam-000000161326823">
                                        <chord xml:id="chord-L12F1" dur="8">
                                            <note xml:id="note-L12F1S1" oct="2" pname="b" accid.ges="f" />
                                            <note xml:id="note-L12F1S2" oct="3" pname="b" accid.ges="f" />
                                        </chord>
                                        <chord xml:id="chord-L14F1" dur="8">
                                            <note xml:id="note-L14F1S1" oct="2" pname="a" accid.ges="n" />
                                            <note xml:id="note-L14F1S2" oct="3" pname="a" accid.ges="n" />
                                        </chord>
                                        <chord xml:id="chord-L16F1" dur="8">
                                            <note xml:id="note-L16F1S1" oct="2" pname="g" accid.ges="n" />
                                            <note xml:id="note-L16F1S2" oct="3" pname="g" accid.ges="n" />
                                        </chord>
                                    </beam>
                                </layer>
                            </staff>
                            <slur xml:id="slur-000000206947535" staff="1" startid="#note-L12F2" endid="#note-L16F2" />
                            <dynam xml:id="dynam-000000012607045" staff="1" tstamp="1.000000">f</dynam>
                            <slur xml:id="slur-000000043774406" staff="1" startid="#note-L9F3" endid="#note-L11F3" curvedir="above" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
Note: if the interval between the notes in two layers is about an octave or more, then adjustment of the augmentation dot placement is probably not needed, since it would be clear which note the augmentation dot belongs to.  Also if one layer has a sustained note and another has a dotted note in the middle of the sustain, the augmentation dot placement would probably not need to be adjusted.

**Comment by rettinghaus** [comment]
@lpugin I want to bring this up again. Is there any chance for this to be fixed?

**Comment by lpugin** [comment]
@rettinghaus sorry, I don't understand. Do you want to fix it? Or are you asking me to fix it? ;-)

**Comment by rettinghaus** [comment]
@lpugin I want you, to do this.

**Comment by rettinghaus** [comment]
fixed with #2122
